### DISCUSSION TITLE
USR-TCP232-E2 - keine Verbindung zum Modbus

### DISCUSSION THREAD

**Discussion by enjoi11** [Tom-Bom-badil/samson_trovis_557x#12] [General (de/en)]
Hallo Tom,

ich nutze den USR-TCP232-E2 und versuche nun seit Tagen zu meiner Trovis 5573 eine Verbindung aufzubauen. Leider ohne Erfolg. Ich habe sämtliche Wiki und Forenthemen durchforstet und abgearbeitet, aber komme einfach nicht weiter. Entweder habe ich etwas grundlegendes übersehen, oder meine Trovis ist ein Sonderfall.

1. Adapter

Adapter Typ: USR-TCP232-E2

2. Trovis

Trovis-Modell:  5573-0003 
Firmware:        2.48

3.  Anbindung

Buchse: seitlich
Kabelbelegung:
1 -> RxD
8 -> TxD
VDD und GND über externes Netzteil (an PIN 2 liegt keine Spannung an)

Der USR-TCP232-E2 ist eingerichtet und ansprechbar (exakt zum Wiki). An der Trovis habe ich ebenfalls die erforderlichen Settings gesetzt C06/F01 = 1 und C06/F03 = 0 (PA6 = 247). Ich schaffe es aber weder mit TrovisView (mit VCOM) oder minimalmodbus eine Verbindung aufzubauen. Ich sehe z. B. im VCOM-Monitor, dass COM+NET Daten ausgetauscht werden, TrovisView gibt aber immer eine Fehlermeldung wie "Zeitüberschreitung bei Empfangen". 

<img width="1137" alt="screenshot" src="https://github.com/Tom-Bom-badil/samson_trovis_557x/assets/26769325/31f27102-dba4-4485-8853-91c7d5d379c1">

Minimalmodbus gibt folgenden Fehler aus:

Checksum error in rtu mode: b'\x01\x90' instead of b'\xd1\x91'

Vielleicht hast du noch eine Idee?

Gruß
enjoi11

**Comment by Tom-Bom-badil** [comment]
Die 5573 wird mir ein ewiges Rätsel bleiben - bereits mehrfach haben Leute sie am Anfang nicht zum Laufen bekommen. Fakt ist, dass es entweder unterschiedliche Buchsenbeschaltungen zu geben scheint, oder bestimmte Adapter nur an bestimmten Pin's der Buchse laufen (kann ich leider nicht testen, da ich keine 5573 habe).

In der [5573-Sektion](https://github.com/Tom-Bom-badil/samson_trovis_557x/wiki/Modellspezifische-Informationen#5573) hier im Wiki sind verschiedene funktionierende Pin-Beschaltungen aufgeführt - vielleicht hilft Dir das weiter. Vermutlich hilft da nur ausprobieren, an welchen Pin's sich die Trovis meldet ...

/tom

Edit/Nachtrag: [Hier](https://knx-user-forum.de/forum/supportforen/smarthome-py/1390281-trovis-557x-heizungsregler-plugin?p=1684988#post1684988) noch ein Beitrag aus dem KNXUF, wo noch ein anderer komischer Effekt beschrieben wurde. Danach lief es dann.

**Comment by enjoi11** [comment]
Danke für die Infos. Ich habe jetzt mal testweise PIN 3+6 angeschlossen. Nun wird mir im Trovis Display "73 SP" angezeigt. Lt. Handbuch erscheint diese Meldung bei Anschluss eines Speichermodul/Minimodul. Der Verbindung klappt aber weiterhin nicht. Die automatische Modemkonfiguration wie aus dem KNXUF ist ebenfalls deaktiviert.

Sind auch noch andere Belegungen möglich als 1+8 bzw. 3+6? Ich würde die ja alle ausprobieren, möchte mir aber ungern das Modul oder den Trovis zerlegen. Auf PIN 2+7 scheinen keine 5V anzuliegen beim Trovis.

**Comment by Tom-Bom-badil** [comment]
Das mit den Pins 3+6 für das Speicher-/Minimodul könnte Licht in das Dunkel bringen, warum dort bei ESP-Anbindungen Daten ankommen. Ich würd dann bei 1+8 bleiben und die ggf. auch mal vertauschen.

Hast Du bei Deinen TrovisView-Versuchen testweise mal von ModbusTCP auf Modbus RTU zurückgestellt?

TTL1 / Socket A / Work Mode (siehe [Anleitung zum K7](https://github.com/Tom-Bom-badil/samson_trovis_557x/wiki/Nachtrag:-Weitere-von-mir-getestete-Adapter#einleitung)):
- TCP Server + None = es wird per RTU kommuniziert, setzt virtuelle COMx: voraus;
- TCP Server + Modbus TCP = es wird per TCP kommuniziert, kein COMx notwendig.

In TrovisView muss natürlich der Verbindungsaufbau entsprechend erfolgen, ist auch oben im Link beschrieben.
Bitte auch vor Verbindungsaufbau nochmal Schnittstelle bzw. Stationsadresse+IP+Port prüfen.

TrovisView erfordert bei mir übrigens gern mal einen Neustart, wenn der erste Verbindungsaufbau nicht erfolgreich war.

**Comment by enjoi11** [comment]
Bei TrovisView und minimalmodbus hatte ich zuletzt nur Modbus RTU genutzt (weil einfacher zu testen mit minimalmodbus).
Gerade eben habe ich mal ModbusTCP getestet - auch hier keine Verbindung. Adressen und Ports sind geprüft.

Und ja, das mit dem Neustart bei TrovisView ist extrem ermüdend :/

Ich glaube mit Pin 1+8 ist irgendeine Kommunikation da (wenn auch vielleicht die falsche). Ich habe auch testweise mal die Modemfunktion aktiviert (ohne die automatische Modemkonfiguration). Dann ist im VCOM Monitor richtig viel los, aber eine Verbindung bekomm ich auch nicht hin.

**Comment by Tom-Bom-badil** [comment]
Modemfunktion aktivieren dürfte nichts bringen - da wird von Modbus auf den [AT-Befehlssatz](https://www.eviateg.de/PDF/Handbuch%20AT-Befehle.pdf#page=13) umgeschaltet, wie ihn 80er/90er Jahre Modems und Faxgeräte verwendet haben (AT --> Ok, ATD (0123)456789 --> Ok <Pfeifen/Piepsen>).

Wenn da 'irgendeine Kommunikation' auf Pin 1+8 stattfindet, klingt das eher nach falschen Kommunikationsparametern (8N1), falscher Baudrate (9600/19200) oder nach einem Port-/Verkabelungsproblem. Der E2 sendet irgendwas raus (legt entsprechende Signale an RxD0/TxD0 an), bei der Trovis kommt aber nix an oder sie kann das Empfangene nicht 'lesen/interpretieren'. Deshalb schickt sie auch nix zurück, was dann zum Timeout auf der Adapterseite führt.

Ich würde wie folgt vorgehen:

- Kommunikationsparameter/Baudrate auf beiden Seiten nochmals prüfen,
- Verkabelung prüfen, ggf. nochmal ein anderes Kabel bauen,
- den anderen Port auf dem E2 ausprobieren (RxD1/TxD1 statt RxD0/TxD0).

/tom

**Comment by darkblu23** [comment]
Hallo enjoi11,
auch ich habe jetzt stundenlang versucht mittels USR-TCP232-E2 meine Trovis 5573-0001 zum sprechen zu bekommen.
Nach vielem hin und her, erst Wiki, dann planlos, ist es mir gelungen die Trovis mit TrovisView zu verbinden.
Ich habe den USR-TCP232-E2 auch so wie du angeschlossen, Rx, Tx aus der Trovis und 5V und GND per USB.
Rx Trovis auf Tx USR-TCP232-E2
Tx Trovis auf Rx USR-TCP232-E2
Das letzte was ich aus Verzweiflung gemacht habe war, dass ich GND von der Trovis auch auf den GND am USR-TCP232-E2 geklemmt habe.
Und siehe da, plötzlich ging die Verbindung zu TrovisView.
Zufall ? Keine Ahnung.
Auf jeden Fall kann TrovisView nun lesen und schreiben.
Und für mich viel wichtiger, Modbus Adapter im ioBroker läuft ebenfalls.

Viel Glück

**Comment by enjoi11** [comment]
Hi @darkblue23,

was soll ich sagen? Es läuft dank deiner Hilfe, ein fettes Danke! ;-) 

Ich habe aber PIN 2 mit auf GND angeschlossen (PIN 1+8 weiterhin für TxD/RxD). 

Siehe dazu auch die Erklärung von robhubi hier: https://github.com/Tom-Bom-badil/samson_trovis_557x/discussions/6#discussioncomment-4368286

<img width="1454" alt="Bildschirmfoto 2023-12-09 um 14 31 10" src="https://github.com/Tom-Bom-badil/samson_trovis_557x/assets/26769325/42b98df2-188c-4ec0-8b02-9caf5afe8b63">

@Tom-Bom-badil vielleicht nimmst du diese Infos in dein Wiki auf? GND scheint ja hier wichtig zu sein bei externer Spannungsversorgung.

VG
enjoi11

**Comment by Tom-Bom-badil** [comment]
Uff, soweit hatte ich gar nicht gedacht - ich komme ursprünglich aus der Elektronik, und deshalb war mir klar, dass zumindest GND durchgeschleift werden muss. Sorry dafür. Technischer Hintergrund (ich versuche es mal zu vereinfachen):

Für die Übertragung elektrischer Signale werden Spannungen verwendet (sagen wir mal: 0.0V ist eine NULL, 3.3V ist eine EINS). Diese Spannungen kann man wie folgt messen:

- Bei 2-Draht (z.B. Modbus über RS485, meist bezeichnet mit A und B): Indem man das Potential auf beiden Adern gegeneinander misst, ergeben sich xxV.
- Bei 3-Draht (z.B. Modbus über RS232 mit RxD / TxD / GND, wie an der Trovis): Hier 'liest' der Modbus-Teilnehmer die Spannung an RxD (Receive Data, Lesen) bzw. legt eine Spannung an TxD (Transmit Data, Senden) an. In beiden Fällen ergibt sich die Spannung aus dem Potential gegenüber GND (Ground / Masse). Man kann sich GND also quasi als gemeinsamen 'Nullpunkt' vorstellen, gegen den das Potential einer Leitung gemessen wird; dafür wird GND auf beiden Seiten durchverdrahtet, um einen gemeinsamen 'Nullpunkt' zu haben.

HINWEIS: Keinesfalls (!!!) gleichzeitig GND und +5V durchverdrahten; NUR (!) GND. Wenn man das z.B. bei einigen ESP-Modellen macht, raucht der interne Spannungskonverter auf dem ESP sogar durch. Solange ich nur über Masse ein gemeinsames Ausgangspotential schaffe, ist alles schick.

Wie geschrieben: Alles stark vereinfacht, aber vielleicht hilft es für das Verständnis.

Glückwunsch zur funktionierenden Anbindung!!!

/tom

_Edit: Hab's mal in die Doku mit aufgenommen, danke für den Hinweis!_

**Comment by enjoi11** [comment]
Danke für nochmal für die Infos bzw. Aufbereitung, dass klingt alles sehr logisch. Wenn ich ehrlich bin, hatte ich solch ein Problem mit GND auch schon im Verdacht. Allerdings wollte ich nicht riskieren, irgendwas zu beschädigen.
